### PR TITLE
[AIRFLOW-2682] Add how-to guides for bash and python operators

### PR DESCRIPTION
**Pull request by tswast**
Make sure you have checked _all_ steps below.

### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-2682

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:

Add a how-to guide page with code examples from the `airflow/example_dags` directory.

image

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

Documentation-only change.

Question: How are the example DAGs tested? I couldn't find documentation in the CONTRIBUTING file on how to add examples or tests for them.

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### Documentation
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

Documentation-only change.

### Code Quality
- [x] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: docs/howto/operator.rst [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+BashOperator
+------------
+
+Use the :class:`~airflow.operators.bash_operator.BashOperator` to execute
+commands in a `Bash <https://www.gnu.org/software/bash/>`__ shell.
+
+.. literalinclude:: ../../airflow/example_dags/example_bash_operator.py
+    :language: python
+    :start-after: [START howto_operator_bash]
```
Comment (ashb) [inline docs/howto/operator.rst:21]: oh nice trick - I didn't know you could do this!

File: docs/howto/operator.rst [hunk 2]
```diff
@@ -0,0 +1,86 @@
+Using Operators
+===============
+
+An operator represents a single, ideally idempotent, task. Operators
+determine what actually executes when your DAG runs.
+
+See the :ref:`Operators Concepts <concepts-operators>` documentation and the
+:ref:`Operators API Reference <api-reference-operators>`for more information.
```
Comment (ashb) [inline docs/howto/operator.rst]: Looks like this ref is wrong - screenshot shows it's not being turned in to a link

Comment (tswast) [reply docs/howto/operator.rst]: Oops, missed some whitespace. Good catch.

Comment (tswast) [reply docs/howto/operator.rst]: Fixed.

image

**Review by Fokko — APPROVED** [review]
Very nice, thanks!

**Pull request closed without merging**